dynamic_programming/climbing_stairs.py

```diff
@@ -3,36 +3,40 @@
 
 def climb_stairs(number_of_steps: int) -> int:
     """
-    LeetCdoe No.70: Climbing Stairs
-    Distinct ways to climb a number_of_steps staircase where each time you can either
-    climb 1 or 2 steps.
+    Calculate the number of distinct ways to climb a staircase with a given number of steps.
+
+    This problem is from LeetCode No. 70: Climbing Stairs.
 
     Args:
-        number_of_steps: number of steps on the staircase
+        number_of_steps (int): The number of steps on the staircase.
 
     Returns:
-        Distinct ways to climb a number_of_steps staircase
+        int: The number of distinct ways to climb the staircase.
 
     Raises:
-        AssertionError: number_of_steps not positive integer
-
-    >>> climb_stairs(3)
-    3
-    >>> climb_stairs(1)
-    1
-    >>> climb_stairs(-7)  # doctest: +ELLIPSIS
-    Traceback (most recent call last):
-        ...
-    AssertionError: number_of_steps needs to be positive integer, your input -7
+        AssertionError: If number_of_steps is not a positive integer.
+
+    Example:
+        >>> climb_stairs(3)
+        3
+        >>> climb_stairs(1)
+        1
+        >>> climb_stairs(-7)  # doctest: +ELLIPSIS
+        Traceback (most recent call last):
+            ...
+        AssertionError: number_of_steps needs to be a positive integer, your input: -7
     """
     assert (
         isinstance(number_of_steps, int) and number_of_steps > 0
-    ), f"number_of_steps needs to be positive integer, your input {number_of_steps}"
+    ), f"number_of_steps needs to be a positive integer, your input: {number_of_steps}"
+
     if number_of_steps == 1:
         return 1
+
     previous, current = 1, 1
     for _ in range(number_of_steps - 1):
         current, previous = current + previous, current
+
```

Comment:
If psf/black does not require changes then we should not perform needless formating of another developer's work.